backend/src/test/java/com/hackner/musicompass/controller/ArtistControllerTest.java

```diff
@@ -0,0 +1,113 @@
+                .discogsArtistUrl(discogsArtistUrl).build();
+        DiscogsArtistSearchResults testDiscogsArtistSearchResults = DiscogsArtistSearchResults.builder()
+                .results(new DiscogsArtist[]{testDiscogsArtist}).build();
+        ResponseEntity<DiscogsArtistSearchResults> mockResponseEntity = ResponseEntity.ok(testDiscogsArtistSearchResults);
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.add("User-Agent", "MusiCompass/0.1");
+        headers.add("Authorization", "Discogs token=" + testAccessToken);
+        HttpEntity<Void> entity = new HttpEntity<>(headers);
+
+        when(restTemplate.exchange(discogsApiUrl, HttpMethod.GET, entity, DiscogsArtistSearchResults.class))
+                .thenReturn(mockResponseEntity);
+        when(testDiscogsSecret.getDiscogsToken()).thenReturn(testAccessToken);
+
+        //WHEN
+        ResponseEntity<Artist> controllerResponse = testRestTemplate.getForEntity(getUrl() + "/" + artistName,Artist.class);
+
+        //THEN
+        assertThat(controllerResponse.getStatusCode(), is(HttpStatus.OK));
+        assertThat(controllerResponse.getBody(), is(Artist.builder()
+                .artistName(artistName)
+                .artistImageUrl(artistImageUrl)
+                .discogsId(discogsArtistId)
+                .discogsArtistUrl(discogsArtistUrl).build()));
+    }
+
+    @Test
+    @DisplayName("Searching for non existing artist name should return 404 error")
+    public void tryToFindNonExistingArtist(){
+        //GIVEN
+        String artistName = "King";
+        String discogsArtistId = "6666";
+        String artistImageUrl = "https://img.discogs.com/wcD5QViPvggOaC3T_D5ql1rHY_E=/150x150/smart/filters:strip_icc():format(jpeg):mode_rgb():quality(40)/discogs-images/A-12596-1565996731-5432.jpeg.jpg";
+        String discogsArtistUrl = "https://api.discogs.com/artists/6666";
+        String baseUrl = "https://api.discogs.com";
+        String discogsApiUrl = baseUrl + "/database/search?type=artist&q=" + artistName;
+        String testAccessToken = "testtest";
+
+        DiscogsArtistSearchResults testDiscogsArtistSearchResults = DiscogsArtistSearchResults.builder()
+                .results(new DiscogsArtist[]{}).build();
+        ResponseEntity<DiscogsArtistSearchResults> mockResponseEntity = ResponseEntity.ok(testDiscogsArtistSearchResults);
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.add("User-Agent", "MusiCompass/0.1");
+        headers.add("Authorization", "Discogs token=" + testAccessToken);
+        HttpEntity<Void> entity = new HttpEntity<>(headers);
+
+        when(restTemplate.exchange(discogsApiUrl, HttpMethod.GET, entity, DiscogsArtistSearchResults.class))
+                .thenReturn(mockResponseEntity);
+        when(testDiscogsSecret.getDiscogsToken()).thenReturn(testAccessToken);
+
+        //WHEN
+        ResponseEntity<Artist> controllerResponse = testRestTemplate.getForEntity(getUrl() + "/" + artistName,Artist.class);
```

Comment:
is this test working? usually you need to indicate ResponseEntity<Void> and Void.class when an error is returned because otherwise the parser tries to parse the expected object from the response

Reply:
test is working on my side. I saw that you used the void class in your "RecipeControllerTest.java". Maybe both class specifications(void and expected class for success) are possible...